joint-discovery-gatk4-fc.wdl

```diff
@@ -0,0 +1,890 @@
+## serve as input to apply the SNP recalibration. If we did it the other way around, 
+## we would have to wait until the SNP recal file was available despite the INDEL 
+## recal file being there already, then apply SNP recalibration, then apply INDEL 
+## recalibration. This would lead to a longer wall clock time for complete workflow 
+## execution. Wiring the INDEL recalibration to be applied first solves the problem.
+##
+## Cromwell version support 
+## - Successfully tested on v29
+## - Does not work on versions < v23 due to output syntax
+##
+## Runtime parameters are optimized for Broad's Google Cloud Platform implementation. 
+## For program versions, see docker containers. 
+##
+## LICENSING : 
+## This script is released under the WDL source code license (BSD-3) (see LICENSE in 
+## https://github.com/broadinstitute/wdl). Note however that the programs it calls may 
+## be subject to different licenses. Users are responsible for checking that they are
+## authorized to run all programs before running this script. Please see the docker 
+## page at https://hub.docker.com/r/broadinstitute/genomes-in-the-cloud/ for detailed
+## licensing information pertaining to the included programs.
+
+workflow JointGenotyping {
+  File unpadded_intervals_file
+
+  String callset_name
+  
+  File ref_fasta
+  File ref_fasta_index
+  File ref_dict
+
+  File dbsnp_vcf
+  File dbsnp_vcf_index
+
+  Array[String] sample_names
+  Array[File] input_gvcfs 
+  Array[File] input_gvcfs_indices 
+
+  String gatk_docker
+
+  Int small_disk
+  Int medium_disk
+  Int huge_disk
+
+  Array[String] snp_recalibration_tranche_values
+  Array[String] snp_recalibration_annotation_values
+  Array[String] indel_recalibration_tranche_values
+  Array[String] indel_recalibration_annotation_values
+
+  File eval_interval_list
+  File hapmap_resource_vcf
+  File hapmap_resource_vcf_index
+  File omni_resource_vcf
+  File omni_resource_vcf_index
+  File one_thousand_genomes_resource_vcf
+  File one_thousand_genomes_resource_vcf_index
+  File mills_resource_vcf
+  File mills_resource_vcf_index
+  File axiomPoly_resource_vcf
+  File axiomPoly_resource_vcf_index
+  File dbsnp_resource_vcf = dbsnp_vcf
+  File dbsnp_resource_vcf_index = dbsnp_vcf_index
+
+  # ExcessHet is a phred-scaled p-value. We want a cutoff of anything more extreme
+  # than a z-score of -4.5 which is a p-value of 3.4e-06, which phred-scaled is 54.69
+  Float excess_het_threshold = 54.69
+  Float snp_filter_level
+  Float indel_filter_level
+  Int SNP_VQSR_downsampleFactor
+
+  Int num_of_original_intervals = length(read_lines(unpadded_intervals_file))
+  Int num_gvcfs = length(input_gvcfs)
+
+  # Make a 2.5:1 interval number to samples in callset ratio interval list
+  Int possible_merge_count = floor(num_of_original_intervals / num_gvcfs / 2.5)
+  Int merge_count = if possible_merge_count > 1 then possible_merge_count else 1
+
+  call DynamicallyCombineIntervals {
```

Comment:
I know this task's purpose is a little confusing so a comment may help here as to what it is trying to do.

Reply:
How about this shortened description of what you wrote previously?: 

"DynamicalyCombineIntervals creates an interval list on the fly from the "golden" 20k even interval_list.  This task now makes it so there is around a 2.5:1 interval:sample ratio.  Its a way to be more efficient by combining intervals  in the original "golden" 20k intervals to get down to a reasonable size for the current callset."

Reply:
Hmm yea the "golden" makes me squirm a little.  I think if we can put the 20k interval list into our standard references bucket we can just say "DynamicalyCombineIntervals combines intervals to an appropriate number for your callset, in this workflow we go for a 2.5:1 interval:sample ratio"

Reply:
I'll make the change once we determine where the 20k interval file will be, if it doesn't need to be standard references bucket I'll do it sooner.